app/controllers/api/projects_controller.rb

```diff
@@ -22,6 +22,37 @@ def searchcode
     render json: Project.visible.where('updated_at > ?', 1.day.ago).order(:repository_url).pluck(:repository_url).compact.reject(&:blank?)
   end
 
+  # returns any updated projects in a time window, the caller can
+  # then decide which ones it needs to refetch.
+  def updated
+    start = date_parameter(:start_time)
+    stop = date_parameter(:end_time)
+
+    raise ActionController::BadRequest.new("must specify start_time") if start.nil?
+    stop = DateTime.current if stop.nil?
+
+    # we have an arbitrary limit on this to prevent pathology, since there's no pagination.
+    arbitrary_maximum = 1000
+
+    count = Project.visible.updated_within(start, stop).count
+
+    # seems better to be loud than to just truncate?
+    raise ActionController::BadRequest.new("query matches too many records") if count > arbitrary_maximum
```

Comment:
Are we envisioning that clients might be aware of this error and adjust their query? If so, I think we might want to return a more structured error.

Reply:
I was kind of going to treat this as a “try to set things up so it rarely happens, and ignore it when it does” type of thing... actually never missing an update would require a more sophisticated overall approach...